File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/resistor-color-trio/resistor-color-trio.js
//
// This is only a SKELETON file for the 'Resistor Color Trio' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class ResistorColorTrio {
  constructor(colors) {
    this.colors = colors;
  }

  get label() {
    const colorMap = {
      black: 0,
      brown: 1,
      red: 2,
      orange: 3,
      yellow: 4,
      green: 5,
      blue: 6,
      violet: 7,
      grey: 8,
      white: 9,
    };

    const [first, second, third] = this.colors;

    if (!colorMap.hasOwnProperty(first) || !colorMap.hasOwnProperty(second) || !colorMap.hasOwnProperty(third)) {
      throw new Error('invalid color');
    }

    const digit1 = colorMap[first];
    const digit2 = colorMap[second];
    const exponent = colorMap[third];

    const mainValue = digit1 * 10 + digit2;
    const totalOhms = mainValue * (10 ** exponent);

    let value = totalOhms;
    let prefix = '';

    if (totalOhms >= 1e9 && totalOhms % 1e9 === 0) {
      value = totalOhms / 1e9;
      prefix = 'giga';
    } else if (totalOhms >= 1e6 && totalOhms % 1e6 === 0) {
      value = totalOhms / 1e6;
      prefix = 'mega';
    } else if (totalOhms >= 1e3 && totalOhms % 1e3 === 0) {
      value = totalOhms / 1e3;
      prefix = 'kilo';
    }

    return prefix ? `${value} ${prefix}ohms` : `${value} ohms`;
  }
}

File: src/2025-02-23-deepclaude-try/java/exercises/practice/resistor-color-trio/src/main/java/ResistorColorTrio.java
class ResistorColorTrio {
    String label(String[] colors) {
        int first = colorCode(colors[0]);
        int second = colorCode(colors[1]);
        int exponent = colorCode(colors[2]);
        
        long[] multipliers = {1L, 10L, 100L, 1000L, 10_000L, 100_000L, 1_000_000L, 10_000_000L, 100_000_000L, 1_000_000_000L};
        long mainValue = (first * 10L) + second;
        long totalOhms = mainValue * multipliers[exponent];
        
        if (totalOhms >= 1_000_000_000L && totalOhms % 1_000_000_000L == 0) {
            return (totalOhms / 1_000_000_000L) + " gigaohms";
        } else if (totalOhms >= 1_000_000L && totalOhms % 1_000_000L == 0) {
            return (totalOhms / 1_000_000L) + " megaohms";
        } else if (totalOhms >= 1_000L && totalOhms % 1_000L == 0) {
            return (totalOhms / 1_000L) + " kiloohms";
        } else {
            return totalOhms + " ohms";
        }
    }
    
    private int colorCode(String color) {
        switch (color) {
            case "black": return 0;
            case "brown": return 1;
            case "red": return 2;
            case "orange": return 3;
            case "yellow": return 4;
            case "green": return 5;
            case "blue": return 6;
            case "violet": return 7;
            case "grey": return 8;
            case "white": return 9;
            default: throw new IllegalArgumentException("Invalid color: " + color);
        }
    }
}
